### PR TITLE
test: deletes notebook test dir

### LINKED ISSUE #31: test: teardown notebook output dir.
Notebook testing creates a test output directory but does not delete it.

### PR DESCRIPTION
**Pull request by ajithopti**
fixes #31

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,7 +87,7 @@ publish-test: release ## publishes artifacts to Test PyPi.
 
 lint: flake
 
-test: PYTEST_ARGS ?= --color=yes --cov-report term --cov=$(SRC)
+test: PYTEST_ARGS ?= --color=yes --cov-report xml --cov-report term --cov=$(SRC)
```
Comment (joetang631) [inline Makefile:90]: I like to use the HTML then XML format. This way I don't have to parse the DOM to get the information, but it depends what is the purpose of the report. If the report is using for visualization tools, then XML might be a better fit. 
An alternative way to consider, I use the HTML plugin, and the output looks very good. Here are the code https://github.com/optimizely/godzilla/blob/60254754661946c165f8d18fd2cc7b4f936aa4b6/run.py#L43
Here is the report looks like:
https://jenkins.dz.optimizely.com/view/Godzilla/job/godzilla_develrc/lastSuccessfulBuild/artifact/log/report.html

File: ssrm_test/test_jupyter_notebooks.py [hunk 2]
```diff
@@ -19,7 +34,10 @@ def generate_papermill_output_dir(tmpdir_factory: object) -> object:
     """
     try:
         os.makedirs(JUPYTER_NOTEBOOK_TESTING_OUTPUT_DIR, exist_ok=True)
-        return JUPYTER_NOTEBOOK_TESTING_OUTPUT_DIR
+        yield JUPYTER_NOTEBOOK_TESTING_OUTPUT_DIR
```
Comment (joetang631) [inline ssrm_test/test_jupyter_notebooks.py:37]: Just curious, any reason we need to use the generator then just return the value? From point of view, in this case, we only create one directory and might not have a performance benefit by using a generator.

File: Makefile (hunk 1 above)
Comment (ajithopti) [reply Makefile:90]: The xml is not for human consumption; it's for uploading to codecov.io or other coverage reporting tools. Also, we still retain the terminal output which us human readable.

File: ssrm_test/test_jupyter_notebooks.py (hunk 2 above)
Comment (ajithopti) [reply ssrm_test/test_jupyter_notebooks.py:37]: The generator pattern is required to include a teardown section to cleanup the test directory created after the test completes. See [here](https://docs.pytest.org/en/stable/fixture.html#fixture-finalization-executing-teardown-code).

**Pull request merged**